include/bio/plain_io/reader.hpp

```diff
@@ -0,0 +1,590 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides bio::plain_io::reader.
+ * \author Hannes Hauswedell <hannes.hauswedell AT decode.is>
+ */
+
+#pragma once
+
+#include <ranges>
+#include <string_view>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/views/to_char.hpp>
+
+#include <bio/plain_io/misc.hpp>
+#include <bio/stream/detail/fast_streambuf_iterator.hpp>
+#include <bio/stream/transparent_istream.hpp>
+
+namespace bio::detail
+{
+
+/*!\brief An input iterator that parses files line-wise and exposes string_views into the read buffer.
+ * \ingroup plain_io
+ * \tparam record_kind_ Whether to read lines or lines+fields.
+ */
+template <plain_io::record_kind record_kind_ = plain_io::record_kind::line>
+class plaintext_input_iterator
+{
+private:
+    //!\brief Down-cast pointer to the stream-buffer.
+    bio::detail::stream_buffer_exposer<char> * stream_buf = nullptr;
+
+    //!\brief Place to store lines that overlap buffer boundaries.
+    std::string         overflow_buffer;
+    //!\brief Temporary storage for field delimiter positions.
+    std::vector<size_t> field_end_positions;
+
+    //!\brief The record.
+    plain_io::record record_;
+
+    //!\brief Whether iterator is at end.
+    bool at_end    = false;
+    //!\brief Delimiter between fields.
+    char field_sep = '\t';
+    //!\brief Delimiter between records [not exposed to modification ATM].
+    char rec_sep   = '\n';
```

Comment:
```suggestion
    char record_sep   = '\n';
```
it's not that much longer and you call it record everywhere else

Reply:
OK